### ISSUE TITLE
od2trips handling impossible trips

### ISSUE THREAD

**Issue by vishalmhjn** [eclipse/sumo#8588]
After reading the docs, I have understood that the od2trips generates a lesser number of trips (e.g., 90-95% although it depends on the network and TAZ file) than the specified because some edges in the TAZs have no possible route. Currently, one option is to specify a greater number than required, so that the resulting number of generated trips are closer, to the initially intended. But I was curious to know if how can a user be more sure to have an exact or more confidence in the number of trips between the zones.

Do you have any plans in future to implement something on the following lines:
1. If the two edges in different TAZs are found to be disconnected, sample another edge from the origin or destination TAZ
2. Check if a trip is possible:
> if YES, go to next trip
> if NO, sample another edge from the origin or destination TAZ

Thanks in advance :)

**Comment by namdre** [comment]
The idea is good but it's currently hard to do because od2trips works without even loading the network and therefore cannot easily check connectivity (somewhat related to #6341).

**Comment by vishalmhjn** [comment]
I agree this (loading network) will compromise the speed of od2trips. I wrote a code (in Python) which recursively processes the TAZ based on the warning output (```error-log```) and filters out the disconnected (source or/and sink) edges from the TAZ file. Although not 100% efficient, this seems to have improved the strike rate of connected trips.
